File: java/leetcode/remove-element.java
class Solution {
    public int removeElement (int[] nums, int val) {
        int i = 0;
        int l = nums.length;
        while (i < l) {
            if (nums[i] == val) {
                nums[i--] = nums[l-- - 1];
            }
            i++;
        }
        return l;
    }
}

File: python/leetcode/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        i = 0
        l = len(nums)
        while i < l:
            if nums[i] == val:
                del nums[i]
                l -= 1
            else:
                i += 1
        return l
